.github/workflows/build-checks.yml

```diff
@@ -21,13 +21,27 @@ jobs:
           - name: clazy
           - name: clang-tidy
           - name: coverage
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-24.04
     name: ${{ matrix.name }}
     steps:
       - name: Check out repository
         uses: actions/checkout@v4.2.1
       - name: Install build dependencies
         run: tools/debian_buildenv.sh setup
+      - name: Build clazy v1.12
+        if: matrix.name == 'clazy'
+        # Ubuntu2404 comes with v1.11, which is too old for the Qt6.4 shipped with Ubuntu 2404
```

Comment:
Do we have a reference for this, a Mixxx and a Ubuntu bug? This sounds like an issue we should solve in our https://github.com/daschuer/mixxx/blob/focal_main/tools/debian_buildenv.sh instead.

Reply:
I could move it to debian_buildenv.sh, but than clazy would be always build and not only for the clazy CI action.

Reply:
The goal should be to have a working clazy locally by default. I looked around to find a PPA for that but not yet found. Maybe we add this to our Launchpad PPA? What is the issue you trying to solve btw.?

Reply:
It can not compile the Qt6.4 headers, because of unsupported C++ features

Reply:
This takes 5 min. A bit long IMHO. 
With a bit of luck the Ocular version will work: https://packages.ubuntu.com/oracular/clazy

Reply:
Can't we just create a docker container with the new clazy? No ideal, but should work okay without delaying every CI build.

Reply:
Using the oracular version does drag a lot of clang dot dependencys in, because they are dependent by an exact version. I consider to push the build to our PPA and make it available for home builds as well.

Reply:
In the meantime I have filed a bug: https://bugs.launchpad.net/ubuntu/+source/clazy/+bug/2086665

Reply:
using the Oracular build works 
``
          echo "deb http://archive.ubuntu.com/ubuntu oracular main universe" | sudo tee /etc/apt/sources.list.d/oracular-temp.list
          echo -e "Package: *\nPin: release n=oracular\nPin-Priority: 100" | sudo tee /etc/apt/preferences.d/oracular.pref
          sudo apt update
          sudo apt install -t oracular clazy
```
Penalty 46s

Reply:
The Oracular hack updates a lot of dependency. Nothing I would like to do at home. 

The macOS solution sounds promising. It looks like macOS builds are in general faster so it is probably a win-win.

Reply:
macOS is not the way to go, the Homebrew installation of Clazy took 1h 50m 50s because it builds the whole llvm package as dependency.

Reply:
Oh, that's bad. So we have two remaining options. Push the clazy build to our PPA, or make use of the GitHub cache. I will look in the PPA solution. It should be straight forward using the Oracular rule file. This has the benefit that Ubuntu users can use it at home as well.

Reply:
We've also the docker container solution proposed by Jan as third option. But I'm also in favor to trying the PPA solution first, as it covers not only CI.